junit-platform-gradle-plugin/src/test/groovy/org/junit/platform/gradle/plugin/JUnitPlatformPluginFunctionalSpec.groovy

```diff
@@ -0,0 +1,272 @@
+			.withArguments('build')
+			.buildAndFail()
+
+		then:
+		result.task(':junitPlatformTest').outcome == TaskOutcome.FAILED
+		result.output.contains('1 tests failed')
+	}
+
+	private static String splitClasspath(List<File> dependencies) {
+		return dependencies
+			.collect { it.absolutePath.replace('\\', '\\\\') } // escape backslashes in Windows paths
+			.collect { "'$it'" }
+			.join(', ')
+	}
+
+	private void javaPlugin() {
+		buildFile << """
+apply plugin: 'java'
+
+sourceCompatibility = JavaVersion.VERSION_1_8
+targetCompatibility = JavaVersion.VERSION_1_8
+
+dependencies {
+	testCompile files(${splitClasspath(testCompileClasspath)})
+	testRuntime files(${splitClasspath(testRuntimeClasspath)})
+	// Use local dependencies so that defaultDependencies are not used
+	junitPlatform files(${splitClasspath(testRuntimeClasspath)})
+}
+
+junitPlatform {
+	filters {
+		engines {
+			include 'junit-jupiter'
+		}
+	}
+}
+"""
+	}
+
+	private void javaLibraryPlugin() {
+		buildFile << """
+apply plugin: 'java-library'
+
+sourceCompatibility = JavaVersion.VERSION_1_8
+targetCompatibility = JavaVersion.VERSION_1_8
+
+dependencies {
+	testImplementation files(${splitClasspath(testCompileClasspath)})
+	testRuntimeOnly files(${splitClasspath(testRuntimeClasspath)})
+	// Use local dependencies so that defaultDependencies are not used
+	junitPlatform files(${splitClasspath(testRuntimeClasspath)})
+}
+
+junitPlatform {
+	filters {
+		engines {
+			include 'junit-jupiter'
+		}
+	}
+}
+"""
+	}
+
+	private void javaFile() {
+		Path javaFile = Paths.get(testProjectDir.root.toString(), 'src', 'main', 'java', 'org', 'junit', 'gradletest', 'Adder.java')
+		Files.createDirectories(javaFile.parent)
+		javaFile.withWriter {
+			it.write(
+				'''
+package org.junit.gradletest;
+
+public class Adder {
+		public int add(int a, int b) {
+				return a + b;
+		}
+}
+''')
+		}
+	}
+
+	private void failingTestFile() {
+		Path testPath = Paths.get(testProjectDir.root.toString(), 'src', 'test', 'java', 'org', 'junit', 'gradletest', 'AdderTest.java')
+		Files.createDirectories(testPath.parent)
+		testPath.withWriter {
+			it.write('''
+package org.junit.gradletest;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+class AdderTest {
+		@Test
+		void failingTest() {
+				Adder adder = new Adder();
+				assertEquals(5, adder.add(3, 3), "This should fail!");
+		}
+}
+'''
+			)
+		}
+	}
+
+	def "when tests are executed again after no changes then the junitPlatformTest task is UP-TO-DATE"() {
+		given:
+		javaPlugin()
+		javaFile()
+		succeedingTestFile()
+
+		when:
+		BuildResult result = GradleRunner.create()
+			.withProjectDir(testProjectDir.root)
+			.withPluginClasspath(pluginClasspath)
+			.withArguments('test')
+			.build()
+
+		then:
+		result.task(':junitPlatformTest').outcome == TaskOutcome.SUCCESS
+
+		when:
+		result = GradleRunner.create()
+												.withProjectDir(testProjectDir.root)
```

Comment:
Good catch, I missed this one.

I typically use spaces and the "chop down and align" settings for continuations.
That is the default for me in IntelliJ.

In this case with tabs the many tab characters lead to something... that looks like this.
I removed a not of the indentation on the others, but missed this one. I'll reduce it to be in the same style as the others.